compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3489,35 +3506,65 @@ TR::Register *OMR::Power::TreeEvaluator::vmulInt32Helper(TR::Node *node, TR::Cod
    lhsReg = cg->evaluate(firstChild);
    rhsReg = cg->evaluate(secondChild);
 
-   resReg = cg->allocateRegister(TR_VRF);
-   tempA = cg->allocateRegister(TR_VRF);
-   tempB = cg->allocateRegister(TR_VRF);
-   tempC = cg->allocateRegister(TR_VRF);
-   node->setRegister(resReg);
-   generateTrg1ImmInstruction(cg, TR::InstOpCode::vspltisw, node, tempA, -16);
-   generateTrg1ImmInstruction(cg, TR::InstOpCode::vspltisw, node, tempB, 0);
-   generateTrg1Src2Instruction(cg, TR::InstOpCode::vmulouh, node, tempC, lhsReg, rhsReg);
-   generateTrg1Src2Instruction(cg, TR::InstOpCode::vrlw, node, resReg, rhsReg, tempA);
-   generateTrg1Src3Instruction(cg, TR::InstOpCode::vmsumuhm, node, tempB, lhsReg, resReg, tempB);
-   generateTrg1Src2Instruction(cg, TR::InstOpCode::vslw, node, tempA, tempB, tempA);
-   generateTrg1Src2Instruction(cg, TR::InstOpCode::vadduhm, node, resReg, tempA, tempC);
-
-   cg->stopUsingRegister(tempA);
-   cg->stopUsingRegister(tempB);
-   cg->stopUsingRegister(tempC);
+   productReg = cg->allocateRegister(TR_VRF);
+   temp = cg->allocateRegister(TR_VRF);
+
+   node->setRegister(productReg);
+
+   generateTrg1ImmInstruction(cg, TR::InstOpCode::vspltish, node, temp, 0);
+   generateTrg1Src3Instruction(cg, TR::InstOpCode::vmladduhm, node, productReg, lhsReg, rhsReg, temp);
+
+   cg->stopUsingRegister(temp);
    cg->decReferenceCount(firstChild);
    cg->decReferenceCount(secondChild);
-   return resReg;
-   }
 
-TR::Register *OMR::Power::TreeEvaluator::vmulFloatHelper(TR::Node *node, TR::CodeGenerator *cg)
-   {
-   return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, TR::InstOpCode::xvmulsp);
+   return productReg;
    }
 
-TR::Register *OMR::Power::TreeEvaluator::vmulDoubleHelper(TR::Node *node, TR::CodeGenerator *cg)
+TR::Register *OMR::Power::TreeEvaluator::vmulInt8Helper(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, TR::InstOpCode::xvmuldp);
+   TR::Node *firstChild;
+   TR::Node *secondChild;
+   TR::Register *lhsReg, *rhsReg;
+   TR::Register *productReg;
+   TR::Register *temp;
+   TR::Register *shiftReg;
+
+   firstChild = node->getFirstChild();
+   secondChild = node->getSecondChild();
+   lhsReg = NULL;
+   rhsReg = NULL;
+
+   lhsReg = cg->evaluate(firstChild);
+   rhsReg = cg->evaluate(secondChild);
+
+   productReg = cg->allocateRegister(TR_VRF);
+   temp = cg->allocateRegister(TR_VRF);
+   shiftReg = cg->allocateRegister(TR_VRF);
+
+   node->setRegister(productReg);
+
+   //set shift amount to 1 byte
+   generateTrg1ImmInstruction(cg, TR::InstOpCode::vspltisb, node, shiftReg, -8);
```

Comment:
I initially did this to keep it consistent with the Int32 implementation (which also used a negative number, -16, as the shift value), but since that's been replaced, I suppose it's no longer necessary.